xds/src/main/java/io/grpc/xds/LookasideLb.java

```diff
@@ -49,169 +48,217 @@
+                GrpcUtil.STOPWATCH_SUPPLIER, Node.getDefaultInstance());
+          }
+        });
+      } else if (xdsClientRefFromResolver != null) {
+        // This is the Non EDS-only usecase.
+        xdsClientRef = xdsClientRefFromResolver;
       } else {
-        channelCredsList = channelCredsListFromResolvedAddresses;
+        // This is the EDS-only with bootstrap usecase.
+        final BootstrapInfo bootstrapInfo;
+        try {
+          bootstrapInfo = bootstrapper.readBootstrap();
+        } catch (Exception e) {
+          lookasideLbHelper.updateBalancingState(
+              TRANSIENT_FAILURE,
+              new ErrorPicker(Status.UNAVAILABLE.withCause(e)));
+          return;
+        }
+        final ManagedChannel channel = initLbChannel(
+            lookasideLbHelper, bootstrapInfo.getServerUri(),
+            bootstrapInfo.getChannelCredentials());
+        xdsClientRef = new RefCountedXdsClientObjectPool(new XdsClientFactory() {
+          @Override
+          XdsClient createXdsClient() {
+            // TODO(zdapeng): Replace XdsComms2 with XdsClientImpl.
+            return new XdsComms2(
+                channel, lookasideLbHelper, new ExponentialBackoffPolicy.Provider(),
+                GrpcUtil.STOPWATCH_SUPPLIER, bootstrapInfo.getNode());
+          }
+        });
       }
 
-      LoadBalancerProvider childBalancerProvider = new LoadBalancerProvider() {
-        @Override
-        public boolean isAvailable() {
-          return true;
-        }
+      // At this point the xdsClientRef is assigned in all usecases, cache them for later use.
+      this.xdsClientRef = xdsClientRef;
+      xdsClient = xdsClientRef.getObject();
+    }
 
-        @Override
-        public int getPriority() {
-          return 5;
-        }
+    // Note: balancerName change is unsupported and ignored.
+    // TODO(zdapeng): Remove support for balancerName.
+    // Note: childPolicy change will be handled in LocalityStore, to be implemented.
 
-        /**
-         * A synthetic policy name identified by balancerName. The implementation detail doesn't
-         * matter.
-         */
-        @Override
-        public String getPolicyName() {
-          return "xds_child_policy_balancer_name_" + newBalancerName;
-        }
+    // If edsServiceName in XdsConfig is changed, do a graceful switch.
+    if (xdsConfig == null
+        || !Objects.equals(newXdsConfig.edsServiceName, xdsConfig.edsServiceName)) {
+      String edsServiceName = newXdsConfig.edsServiceName;
 
-        @Override
-        public LoadBalancer newLoadBalancer(final Helper helper) {
-          return new LoadBalancer() {
-            @Nullable
-            XdsClient xdsClient;
-            @Nullable
-            LocalityStore localityStore;
-            @Nullable
-            LoadReportClient lrsClient;
+      // Just in case edsServiceName is null, in the future this shouldn't be allowed.
```

Comment:
(Good to have) You could just say: legacy gRPC client with EDS uses target authority for querying endpoints, but in the future we expect this to be explicitly given by EDS config.

Reply:
Done.